contracts/utils/RetireToucanCarbon.sol

```diff
@@ -0,0 +1,1405 @@
+                        redeemERC20,
+                        redeemAmount
+                    );
+
+                    // Retire TCO2
+                    IToucanCarbonOffsets(_listTCO2[i]).retire(poolBalance);
+                    emit ToucanRetired(
+                        msg.sender,
+                        _beneficiary,
+                        _toucanPool,
+                        _listTCO2[i],
+                        poolBalance
+                    );
+
+                    leftToBurn -= poolBalance;
+                } else {
+                    // Redeem from pool
+                    redeemAmount[0] = leftToBurn;
+                    IBaseCarbonTonne(_toucanPool).redeemMany(
+                        redeemERC20,
+                        redeemAmount
+                    );
+
+                    // Retire TCO2
+                    IToucanCarbonOffsets(_listTCO2[i]).retire(leftToBurn);
+                    emit ToucanRetired(
+                        msg.sender,
+                        _beneficiary,
+                        _toucanPool,
+                        _listTCO2[i],
+                        leftToBurn
+                    );
+
+                    leftToBurn = 0;
+                }
+            }
+        }
+
+        require(
+            leftToBurn == 0,
+            "Not all pool tokens were burned. Please submit a larger TCO2 token list."
+        );
+    }
+
+    function _bondFees(address _toucanPool) internal returns (bool) {
+        // Bond cumulative fees to the DAO if bond is large enough
+        uint256 poolBalance = IERC20(_toucanPool).balanceOf(address(this));
+
+        uint256 bondPrice = IKlimaBondDepository(bondDepository[_toucanPool])
+            .bondPriceInUSD();
+
+        if ((poolBalance * (10**9)) / bondPrice > 10000000) {
+            IERC20(_toucanPool).safeIncreaseAllowance(
+                bondDepository[_toucanPool],
+                poolBalance
+            );
+
+            IKlimaBondDepository(bondDepository[_toucanPool]).deposit(
+                poolBalance,
+                bondPrice,
+                DAO
+            );
+        }
+        return true;
+    }
+
+    function setFeeAmount(uint256 _amount) external onlyOwner returns (bool) {
+        emit FeeUpdated(feeAmount, _amount);
+
+        feeAmount = _amount;
+        return true;
+    }
+
+    function setPoolRouter(address _toucanPool, address _router)
+        external
+        onlyOwner
+        returns (bool)
+    {
+        poolRouter[_toucanPool] = _router;
+        return true;
+    }
+
+    function addPool(
+        address _toucanPool,
+        address _router,
+        address _bondDepository
+    ) external onlyOwner returns (bool) {
+        bool result;
+
+        require(!listContains(toucanPools, _toucanPool), "Pool already added");
+
+        if (!listContains(toucanPools, _toucanPool)) {
+            toucanPools.push(_toucanPool);
+        }
+        result = !isToucanPool[_toucanPool];
```

Comment:
This will be always false at this point so I think it can be removed altogether and you can just:
```js
isToucanPool[_toucanPool] = true;
```

Reply:
Implemented